File: market-intelligence/src/main/java/com/hope/trading/market_intelligence/adapter/marketdata/MarketResponse.java
package com.hope.trading.market_intelligence.adapter.marketdata;

import java.time.Instant;
import java.util.UUID;

public record MarketResponse(
        UUID marketId,
        String provider,
        String symbol,
        String baseAsset,
        String quoteAsset,
        MarketStateResponse marketState
) {
    public record MarketStateResponse(
            String tradingStatus,
            boolean tradable,
            String closureReason,
            Instant lastUpdated
    ) {
    }
}

File: trading-os-web/src/app/core/models/market-response.ts
export interface MarketResponse {
  marketId: string;

  provider: string;

  symbol: string;

  baseAsset: string;

  quoteAsset: string;

  marketState: {
    tradingStatus: string;
    tradable: boolean;
    closureReason: string;
    lastUpdated: string;
  };

  marketConstraints: {
    minimumOrderSize: number;
    minimumCost: number;
    tickSize: number;
    quantityPrecision: number;
    pricePrecision: number;
  };
}
